modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -157,12 +158,7 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
    */
   material(material: Material): this {
     this.__material = material;
-    this._registry = getRegistry({
-      chainName: material.chainName,
-      specName: material.specName,
-      specVersion: material.specVersion,
-      metadataRpc: material.metadata,
-    });
+    this._registry = SingletonRegistry.getInstance(material);
```

Comment:
How does this reduce the memory usage for the transaction builder method? Correct me if I am wrong, but aren't we still storing this in an instance the same way we do here ^? Still having 2 materials stored in this builder

Reply:
Material is a string and since JS has string interning (https://en.wikipedia.org/wiki/String_interning) there are no two instances of it. Memory is consumed only once, parsing the metadata is what consumes memory so we should be careful about that.